lgc/builder/DescBuilder.cpp

```diff
@@ -403,6 +403,17 @@ Value *DescBuilder::CreateGetBufferDescLength(Value *const bufferDesc, Value *of
   return CreateNamedCall(lgcName::LateBufferLength, getInt32Ty(), {bufferDesc, offset}, Attribute::ReadNone);
 }
 
+// =====================================================================================================================
+// Return the i64 difference between two buffer fat pointer values, dividing out the size of the pointed-to objects.
+//
+// @param lhs : Left hand side of the subtraction.
+// @param rhs : Reft hand side of the subtraction.
+// @param instName : Name to give instruction(s)
+Value *DescBuilder::CreateBufferPtrDiff(llvm::Value *lhs, llvm::Value *rhs, const llvm::Twine &instName) {
```

Comment:
Could this be "CreatePtrDiff", overriding the one in IRBuilder, and calling the one in IRBuilder if the pointers are not fat pointers?

I am a bit dubious about relying on the pointed-to type of the pointers, as LLVM is moving away from that. But that's what IRBuilder does, and I guess we should do the same until IRBuilder does something new.

Reply:
Rewrote it to redefine `CreatePtrDiff`. This somehow also fixed the CTS failures.

> I am a bit dubious about relying on the pointed-to type of the pointers, as LLVM is moving away from that. But that's what IRBuilder does, and I guess we should do the same until IRBuilder does something new.

True, I’m interested in how LLVM is going to handle that.

Reply:
> This somehow also fixed the CTS failures.

:thumbsup:

The failures were in tests that exercise PtrDiff with variables in Workgroup storage class (so not buffer). I think what happened in the first version was lgc.late.buffer.ptrdiff got left behind after PatchBuffer pass.